### DISCUSSION TITLE
Use Main Layout left drawer to pass data into router-vue

### DISCUSSION THREAD

**Discussion by Davdup** [quasarframework/quasar#8900] [General - Components / Directives / etc] [answered]
Hi,
I'm quite new to Quasar and Vue.js, and my question is probably quite basic...
I would like to put some toggles, hooked to three v-models (option1 / option2 /option3) in a left drawer (which is defined in the layouts/Mainlayout.vue file), and use these three parameters in the main page, injected where the <router-view/> is placed in the same layouts/Mainlayout.vue file. Let's say that the main page is mainPage.vue
I guess I have to set-up something in the export default section of the two vue files.
I know how to pass props between a main page and a components, but I don't know how to do this when using the <router-view/> injection...?
Any Help??
Thank you very much!

David

**Comment by metalsadman** [comment, marked as answer]
I don't see anything wrong passing it as a props when it's the normal communication from parent to child. about injection, read about it in vue docs.

**Reply by Davdup** [reply]
Hi, thanks for your reply! I found a solution by using a Vuex store, that will be able to keep all these parameters and make them accessible anywhere I guess... anyway that works for me!

**Reply by metalsadman** [reply]
> Hi, thanks for your reply! I found a solution by using a Vuex store, that will be able to keep all these parameters and make them accessible anywhere I guess... anyway that works for me!

yeah that's other way too, please mark the thread as answered, thx.
